dataland-backend/src/main/kotlin/org/dataland/datalandbackend/configurations/SecurityConfig.kt

```diff
@@ -0,0 +1,77 @@
+package org.dataland.datalandbackend.configurations
+
+import org.keycloak.adapters.springsecurity.KeycloakConfiguration
+import org.keycloak.adapters.springsecurity.config.KeycloakWebSecurityConfigurerAdapter
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Profile
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.core.authority.mapping.SimpleAuthorityMapper
+import org.springframework.security.web.authentication.session.NullAuthenticatedSessionStrategy
+import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy
+import org.springframework.stereotype.Component
+
+/**
+ * This class contains all security configurations for the backend to secure endpoints with keycloak.
+ */
+@Profile("!unprotected")
+@KeycloakConfiguration
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+class SecurityConfig : KeycloakWebSecurityConfigurerAdapter() {
+
+    private val publicLinks = arrayOf(
+        "/actuator/health",
+        "/actuator/health/ping",
+        "/actuator/info",
+        "/swagger-ui/**",
+        "/v3/api-docs/**",
+        "/companies/**",
+        "/data/**",
+        "/metadata",
+        "/metadata/**"
+    )
+
+    /**
+     * The Role Container is a component which maps actual roles from the Dataland keycloak service to the role
+     * definitions which are used in the Dataland backend code to decouple the role namings in the keycloak service
+     * from the ones in backend.
+     */
+    @Component("RoleContainer")
+    object RoleContainer {
+        const val DATA_READER = "USER"
```

Comment:
Als env auslagern?

Reply:
Dient quasi als "Mapping" zwischen dem Rollennamen in KeyCloak und im Code.
Daher würde ich eher nicht dazu tendieren, es als env auszulagern.
Was meinst du? Siehst du Argumente dafür oder dagegen?

Reply:
Ich würde es erstmal so lassen.